### ISSUE TITLE
Claude-generated: solver="auto" silently ignores IPM-only settings (smoothed differentiation, yolo)

### ISSUE THREAD

**Issue by sbarratt** [moreau-project/moreau#66]
Claude-generated. Found by an external review of 0.4.1 and re-verified on main @ e794100 (CPU).

## Problem

When `solver="auto"` resolves to active-set (small QPs), explicit IPM-only settings are silently ignored:

- `IPMSettings(diff_method="smoothed", ...)`, exactly as shown in `docs/guide/smoothed-differentiation.md`, returns the exact active-set gradient instead of the smoothed one.
- `yolo=True` returns `Solved` after 1 iteration instead of running `yolo_num_iters` fixed IPM iterations.
- The same applies to IPM tolerances.

```python
import numpy as np, scipy.sparse as sp, torch, moreau
from moreau.torch import Solver
def grad(settings):
    s = Solver(n=1, m=1, P_row_offsets=torch.tensor([0, 1]), P_col_indices=torch.tensor([0]),
               A_row_offsets=torch.tensor([0, 1]), A_col_indices=torch.tensor([0]),
               cones=moreau.Cones(num_nonneg_cones=1), settings=settings)
    q = torch.tensor([-1.0], dtype=torch.float64, requires_grad=True)
    one = torch.ones(1, dtype=torch.float64)
    s.solve(one, one, q, one).x.sum().backward()
    return float(q.grad)
sm = moreau.IPMSettings(diff_method="smoothed", diff_smoothing_mu=1e-1)
print("auto:", grad(moreau.Settings(ipm_settings=sm)), " ipm:", grad(moreau.Settings(solver="ipm", ipm_settings=sm)))
P = sp.csr_array(np.eye(2)); A = sp.csr_array(np.vstack([np.ones((1, 2)), -np.eye(2)])); b = np.r_[1.0, 0, 0]
for solver in ("auto", "ipm"):
    s = moreau.Solver(P, np.ones(2), A, b, moreau.Cones(num_zero_cones=1, num_nonneg_cones=2),
                      settings=moreau.Settings(solver=solver, yolo=True, yolo_num_iters=10))
    s.solve(); print(f"yolo solver={solver}: {s.info.status.name}, {s.info.iterations} iterations")
```

```
auto: -1.0  ipm: -0.4999996664051878
yolo solver=auto: Solved, 1 iterations
yolo solver=ipm: MaxIterations, 10 iterations
```

## Scope

- [ ] When the user sets IPM-only options (non-default `ipm_settings`, `yolo=True`), have `auto` resolve to IPM, or at least warn that they are ignored.
- [ ] Test both cases.

Related: #48.
